### PR TITLE
Extend interface device definitions

### PR DESCRIPTION
**Pull request by vincentbernat**
This is still incomplete but most of the useful things are
here. Examples are taken from libvirt samples.

### CODE REVIEW

File: domain.go [hunk 1]
```diff
@@ -62,13 +62,55 @@ type DomainInterfaceMAC struct {
...
+}
+
+type DomainInterfaceSource struct {
+	Bridge  *string `xml:"bridge,attr"`
+	Network *string `xml:"network,attr"`
+	Address *string `xml:"address,attr"`
+	Type    *string `xml:"type,attr"`
+	Path    *string `xml:"path,attr"`
+	Mode    *string `xml:"mode,attr"`
+	Port    *uint   `xml:"port,attr"`
```
Comment (berrange) [inline domain.go]: These shouldn't be using pointers as it makes it tedious to initialize. The style used elswhere is to have omitempty annotations.

File: domain.go [hunk 2]
```diff
@@ -62,13 +62,55 @@ type DomainInterfaceMAC struct {
...
+	Order int `xml:"order,attr"`
+}
+
+type DomainInterfaceScript struct {
+	Path string `xml:"path,attr"`
+}
+
+type DomainInterfaceDriver struct {
+	Name   string `xml:"name,attr"`
+	Queues *uint  `xml:"queues,attr"`
```
Comment (berrange) [inline domain.go]: Again, no pointers

File: domain_test.go [hunk 3]
```diff
@@ -307,19 +307,243 @@ var domainTestData = []struct {
...
 		if err != nil {
-			t.Fatal(err)
+			t.Error(err)
 		}
 
 		expect := strings.Join(test.Expected, "\n")
 
 		if doc != expect {
-			t.Fatal("Bad xml:\n", string(doc), "\n does not match\n", expect, "\n")
+			t.Error("Bad xml:\n", string(doc), "\n does not match\n", expect, "\n")
```
Comment (berrange) [inline domain_test.go]: These changes are unrelated to the rest of the commit - please put them in a separate commit and why why they're needed

**Review by berrange — CHANGES_REQUESTED** [review]

**Comment by vincentbernat** [comment]
I have used `omitempty` instead of pointers. I have removed the last unrelated change.

**Comment by berrange** [comment]
Thanks this is merged

**Pull request closed without merging**